scipy/special/special/specfun/specfun.h

```diff
@@ -4777,7 +4777,7 @@ inline void rmn1(int m, int n, T c, T x, int kd, T *df, T *r1f, T *r1d) {
 
     cx = c * x;
     nm2 = 2 * nm + m;
-    sphj(static_cast<T>(nm2), cx, &nm2, sj, dj);
+    sphj(cx, nm2, &nm2, sj, dj);
```

Comment:
The argument order of `sphj` was somehow switched during the translation, but it was still being called with the old order here.

https://github.com/scipy/scipy/blob/1d3a067c2ccd0a6efddeb3194163aa9a3879d26e/scipy/special/specfun/specfun.f#L9683

to

https://github.com/scipy/scipy/blob/1bd1e451561af13a76c66d84b8003391322d5e1e/scipy/special/special/specfun/specfun.h#L5572

Reply:
Not sure how this passed the compilation step. Probably we are not as strict as we should be with the compiler flags.

Reply:
yeah, that's definitely something we should look into to